docs/docs/guides/infer_contract_types_guide/index.md

```diff
@@ -0,0 +1,112 @@
+
+```typescript
+import { Contract, Web3 } from 'web3';
+import ERC20 from './node_modules/@openzeppelin/contracts/build/contracts/ERC20.json';
+
+(async function () {
+	const web3 = new Web3('rpc url');
+
+	const contract = new Contract(ERC20.abi, '0x7af963cF6D228E564e2A0aA0DdBF06210B38615D', web3);
+
+	const holder = '0xa8F6eB216e26C1F7d924A801E46eaE0CE8ed1A0A';
+
+	//Error because Contract doesn't know what methods exists
+	const balance = await contract.methods.balanceOf(holder).call();
+})();
+```
+
+To work around it you need to copy abi into a Typescript file like this:
+
+```typescript
+import {Contract, Web3} from "web3";
+
+
+const ERC20 = [
+    ...
+    // "as const" is important part, without it typescript would create generic type and remove available methods from type
+] as const;
+
+(async function() {
+    const web3 = new Web3("rpc url")
+
+    const contract = new Contract(ERC20, "0x7af963cF6D228E564e2A0aA0DdBF06210B38615D", web3)
+
+    const holder = "0xa8F6eB216e26C1F7d924A801E46eaE0CE8ed1A0A"
+
+    //Works now
+    const balance = await contract.methods.balanceOf(holder).call()
+})()
+```
+
+Now it's working but it also means that abi is no longer updated when you bump your npm dependencies.
+To solve this problem, you can use a custom script that copies the JSON artifact of the contract into a TypeScript file as a const variable. This script can be run as part of your build process so that the TypeScript file is always up-to-date with the latest version of the contract's ABI.
+
+Script:
+
+```typescript title="gen.ts"
+#!/usr/bin/node
```

Comment:
We don't have to include this line right? This would be invalid for anyone using nvm (which I believe is best practice nowadays)